### ISSUE TITLE
`v1` support and the future of `tektoncd/catalog`

### ISSUE THREAD

**Issue by vdemeester** [tektoncd/catalog#1146]
This issue is to discuss and track work around having `v1` API support for tasks in the catalog, in the light of the recent TEP around catalog: 
- [TEP-0079: Tekton Catalog Support Tiers](https://github.com/tektoncd/community/blob/main/teps/0079-tekton-catalog-support-tiers.md)
- [TEP-0115: Tekton Catalog Git-Based Versioning](https://github.com/tektoncd/community/blob/main/teps/0115-tekton-catalog-git-based-versioning.md)

As of [`tektoncd/pipeline`](https://github.com/tektoncd/pipeline) latest LTS release (0.44.x), we provide `v1` API for tekton pipeline's objects. This means, we can start publishing `Task`s using `v1`. *But* as TEP-0079 state, we want to move towards having smaller, well written and supported `Task`s (and other resource) maintained, in the [`tektoncd-catalog`](https://github.com/tektoncd-catalog) org. And let contributors share and maintain resources in their own Community Catalogs. Hence this issue.

1. All the task that are migrating to a `tektoncd-catalog` repository will switch to `v1` there and marked as deprecated here.
2. For the rest of Task, we probably want to notify OWNERS and/or adopt those in our own set of catalog (our meaning "individuals" or " companies").
3. What deadline do we set for marking this repository as deprecated as well ?

/cc @tektoncd/catalog-maintainers @tektoncd/catalog-collaborators @QuanZhang-William @bobcatfish @jerop

**Comment by QuanZhang-William** [comment]
Thanks @vdemeester! For , taking the example of golang catalog migration, the current selection of the resources to be migrated is in `v1beta1`, and we planned to version it as [v1.0.0 in the new catalog](https://github.com/tektoncd/community/blob/main/teps/0115-tekton-catalog-git-based-versioning.md#tekton-catalog-github-organization).

If we want to bump the API version to `v1`, do we directly make the change in the `v1.0.0` release of the catalog or creating a new patch version like `v1.0.1`?

**Comment by vinamra28** [comment]
yes @QuanZhang-William, we should be bumping up the version of the resource

**Comment by vdemeester** [comment]
@QuanZhang-William I think it really depends on what we want to support. We could state that anything going in `tektoncd-catalog` is going to be `v1` only and we could start `v1.0.0` with that. Otherwise, we would still need to bump more than just a patch version (as it's using a new API).

**Comment by QuanZhang-William** [comment]
> @QuanZhang-William I think it really depends on what we want to support. We could state that anything going in `tektoncd-catalog` is going to be `v1` only and we could start `v1.0.0` with that. Otherwise, we would still need to bump more than just a patch version (as it's using a new API).

Agreed. I think the question is that do we want to continue to maintain `v1beta1` resources in the verified catalogs. If we don't plan to make changes to `v1beta1` resources in the verified catalogs, I think it makes more sense to directly start with `v1` since the `v1beta1` resources are still available to users in the old central catalog even though it will be archived soon.

Supporting both `v1beta1` and `v1` resources also  doubles the maintenance cost. For the [git-clone](https://github.com/tektoncd/catalog/tree/main/task/git-clone) example: if we support both API versions, we need to migrate git-clone v0.6 - v0.9 in the old catalog (using `v1beta1` API version) to `v1.0.0` to `v4.0.0` in the new catalog, then bump the API version to `v1` and continue to create catalog release `v5.0.0` to `v8.0.0`? This sounds a bit weird to me.

WDYT?

/cc @ywluogg

**Comment by QuanZhang-William** [comment]
Catalog WG summary: The verified catalogs will be started with `v1` api version.
